src/main/resources/mapper/RouteMapper.xml

```diff
@@ -28,8 +28,8 @@
         SELECT id, start_point, end_point, distance FROM routes
     </select>
 
-    <insert id="setWayPoints">
-        INSERT INTO waypoints (points_id, routes_id) VALUES (#{points_id}, #{routes_id})
+    <insert id="setWayPoints" parameterType="map">
+        INSERT INTO waypoints (points_id, routes_id) VALUES (#{point.id}, #{route.id})
```

Comment:
think this should be points.id and routes.id (missing 's') if we're referencing the tables